src/bindings/Node.re

```diff
@@ -0,0 +1,320 @@
+  /*   "writeFile"; */
+  /* let writeFile = promisify2(writeFile); */
+
+  /* [@bs.module "fs"] */
+  /* external readFile: */
+  /*   (string, (. Js.Nullable.t(Error.t), string) => unit) => unit = */
+  /*   "readFile"; */
+  /* let readFile = promisify(readFile); */
+
+  module E = {
+    type t =
+      | PathNotFound;
+    let toString =
+      fun
+      | PathNotFound => "mkdir(~p=true) received home path and it was not found";
+  };
+
+  module Stat = {
+    type t;
+    [@bs.send] external isDirectory: t => bool = "isDirectory";
+  };
+
+  [@bs.module "fs"]
+  external stat': (. string, (Js.Nullable.t(Error.t), Stat.t) => unit) => unit =
+    "stat";
+
+  let stat = p =>
+    Js.Promise.make((~resolve, ~reject as _) =>
+      stat'(. p, (err, stat) =>
+        resolve(.
+          switch (Js.Nullable.toOption(err)) {
+          | Some(_e) => Error("Couldn't stat")
+          | None => Ok(stat)
+          },
+        )
+      )
+    );
+
+  [@bs.module "fs"]
+  external readDir':
+    (. string, (Js.Nullable.t(Error.t), array(string)) => unit) => unit =
+    "readdir";
+  let readDir = path =>
+    Promise.make((~resolve, ~reject as _) => {
+      readDir'(. path, (error, files) => {
+        switch (Js.Nullable.toOption(error)) {
+        | Some(error) => resolve(. Error(error##message))
+        | None => resolve(. Ok(files))
+        }
+      })
+    });
+
+  type fd;
+  [@bs.module "fs"] external writeSync: (. fd, Buffer.t) => unit = "writeSync";
+
+  [@bs.module "./fs-stub.js"]
+  external writeFile: (string, string) => Js.Promise.t(unit) = "writeFile";
+
+  [@bs.module "./fs-stub.js"]
+  external readFile: string => Js.Promise.t(string) = "readFile";
+
+  [@bs.module "./fs-stub.js"]
+  external mkdir': string => Js.Promise.t(unit) = "mkdir";
+
+  [@bs.module "./fs-stub.js"]
+  external exists: string => Js.Promise.t(bool) = "exists";
+
+  [@bs.module "./fs-stub.js"]
+  external open_: (string, string) => Js.Promise.t(fd) = "open";
+
+  [@bs.module "./fs-stub.js"]
+  external write: (fd, Buffer.t) => Js.Promise.t(unit) = "write";
+
+  [@bs.module "./fs-stub.js"]
+  external close: (fd, Buffer.t) => Js.Promise.t(unit) = "close";
+
+  [@bs.module "fs"]
+  external createWriteStream: string => Stream.t = "createWriteStream";
+
+  [@bs.module "./fs-stub.js"]
+  external unlink: string => Js.Promise.t(bool) = "unlink";
+
+  let rec mkdir = (~p=?, path) => {
+    let forceCreate =
+      switch (p) {
+      | Some(x) => x
+      | None => false
+      };
+    Js.Promise.(
+      if (forceCreate) {
+        exists(path)
+        |> then_(doesExist =>
+             if (doesExist) {
+               resolve(Ok());
+             } else {
+               let homePath = Sys.getenv(Sys.unix ? "HOME" : "USERPROFILE");
+               if (path == homePath) {
+                 resolve(Error(E.PathNotFound));
+               } else {
+                 mkdir(~p=true, Filename.dirname(path))
+                 |> then_(
```

Comment:
Do you think this code is easy to understand? There's like 6 levels of nested if statements. I certainly don't write code like this.